src/main/java/org/folio/dao/lines/PoLinesPostgresDAO.java

```diff
@@ -61,18 +62,19 @@ public Future<PoLine> getPoLineById(String id, DBClient client) {
   }
 
   @Override
-  public Future<Integer> updatePoLines(String sql, DBClient client) {
+  public Future<Integer> updatePoLines(String sql, Conn conn) {
     log.debug("updatePoLines, sql={}", sql);
     Promise<Integer> promise = Promise.promise();
-    client.getPgClient().execute(sql, ar -> {
-      if (ar.failed()) {
-        log.error("updatePoLines failed, sql={}", sql, ar.cause());
-        handleFailure(promise, ar);
-      } else {
-        log.debug("updatePoLines success, sql={}", sql);
-        promise.complete(ar.result().rowCount());
-      }
-    });
+    conn.execute(sql)
+      .onComplete(ar -> {
+        if (ar.failed()) {
```

Comment:
Can we use here onSuccess and onFailure instead of onComplete?

Reply:
I am not sure which one is better, so I usually keep the existing code as much as possible without changing that. Here the existing logic seemed to match `onComplete` better. Also I tend to prefer `onComplete` when we need code for both success and failure, to use a single terminal handler and because I suspect it has slightly better performance (when we need to test both success and failure). Also, when would you ever use `onComplete` otherwise ? Interestingly, in Scala, `onSuccess` and `onFailure` were deprecated in favor of onComplete (I am not sure why).
Why do you prefer `onSuccess` and `onFailure` in this case ?

Reply:
As for me using onSuccess and onFailure handlers can make code clearer, reduce the chance of errors(we don't need custom error handler), and simplify the handling of different operation outcomes in an organized manner.